environment.yml

```diff
@@ -21,7 +21,7 @@ dependencies:
   - humanfriendly
   - intake-esgf >=2025.10.22
   - intake-esm
-  - iris >=3.12.2,<3.14.1 # https://github.com/SciTools/iris/issues/6417, https://github.com/ESMValGroup/ESMValCore/issues/2921
+  - iris >=3.14.1
```

Comment:
Do we need this pin? Just removing the upper limit may be sufficient and it gives users some room to choose versions

Reply:
problem is 3.14 was a bit iffy: they momentarily removed Python 3.14 support in that, maybe we can just `!=3.14`?